dolphinscheduler-datasource-plugin/dolphinscheduler-datasource-kyuubi/src/main/java/org/apache/dolphinscheduler/plugin/datasource/kyuubi/KyuubiDataSourceClient.java

```diff
@@ -44,9 +48,13 @@ protected void preInit() {
     @Override
     protected void initClient(BaseConnectionParam baseConnectionParam, DbType dbType) {
 
-        this.dataSource = JDBCDataSourceProvider.createOneSessionJdbcDataSource(baseConnectionParam, dbType);
-        this.jdbcTemplate = new JdbcTemplate(dataSource);
-        log.info("Init {} success.", getClass().getName());
+        this.driverManagerDataSource =
+                new DriverManagerDataSource(DataSourceUtils.getJdbcUrl(DbType.KYUUBI, baseConnectionParam),
+                        baseConnectionParam.getUser(), PasswordUtils.decodePassword(baseConnectionParam.getPassword()));
+        driverManagerDataSource.setDriverClassName(baseConnectionParam.getDriverClassName());
+        this.jdbcTemplate = new JdbcTemplate(driverManagerDataSource);
+        log.info("Init {} {} success.", getClass().getName(),
+                DataSourceUtils.getJdbcUrl(DbType.KYUUBI, baseConnectionParam));
```

Comment:
We don't need to use `DriverManagerDataSource` here, we can directly generate new connections.

Reply:
DriverManagerDataSource should create a new connection each time, so we possibly can stick to that.